Sources/Stats/py_stats.cc

```diff
@@ -127,6 +129,77 @@ void AddStatsModule(py::module m) {
                     Rank-2 tensors should have shape `(N, M)` where `N` is the
                     number of samples in one Markov Chain and `M` is the number
                     of Markov Chains. Data should be in row major order.)EOF");
+
+  subm.def(
+      "covariance_sv",
+      [](py::array_t<Complex, py::array::c_style> s_values,
+         py::array_t<Complex, py::array::c_style> v_values, bool center_s) {
+        Eigen::Map<const VectorXcd> s_vector{s_values.data(), s_values.size()};
+
+        // Compute S -> S - 𝔼[S]
+        const Complex mean = [&s_vector, center_s]() -> Complex {
+          if (center_s) {
+            Complex mean = s_vector.mean();
+            MeanOnNodes(mean);
+            return mean;
+          } else {
+            return 0.;
+          }
+        }();
+
+        switch (s_values.ndim()) {
+          case 2:
+            NETKET_CHECK(v_values.ndim() == 3, InvalidInputError,
+                         "v_values has wrong dimension: " << v_values.ndim()
+                                                          << "; expected 3.");
+            return Gradient(
+                s_vector.array() - mean,
+                Eigen::Map<const RowMatrix<Complex>>{
+                    v_values.data(), v_values.shape(0) * v_values.shape(1),
+                    v_values.shape(2)});
+          case 1:
+            NETKET_CHECK(v_values.ndim() == 2, InvalidInputError,
+                         "v_values has wrong dimension: " << v_values.ndim()
+                                                          << "; expected 2.");
+            return Gradient(
+                s_vector.array() - mean,
+                Eigen::Map<const RowMatrix<Complex>>{
+                    v_values.data(), v_values.shape(0), v_values.shape(1)});
+          default:
+            NETKET_CHECK(false, InvalidInputError,
+                         "s_values has wrong dimension: "
+                             << s_values.ndim() << "; expected either 1 or 2.");
+        }  // end switch
+      },
+      py::arg{"s"}.noconvert(), py::arg{"v"}.noconvert(),
+      py::arg{"center_s"} = true,
+      R"EOF(Computes the covariance of two random variables S, V from MCMC
+            data. Note that while S has to be a complex scalar variable,
```

Comment:
would this work for real-valued S or would you get an error?

Reply:
It would throw an error, like this:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-19-586e4b786a84> in <module>
----> 1 x = nk.stats.covariance_sv(elocs.real, derlogs)

TypeError: covariance_sv(): incompatible function arguments. The following argument types are supported:
    1. (s: numpy.ndarray[complex128], v: numpy.ndarray[complex128], center_s: bool = True) -> numpy.ndarray[complex128[m, 1]]

```

Reply:
OK, but the natural behavior would be to accept also floats and compute the standard covariance in that case... We can change this later I guess, but it would make sense